exasol_advanced_analytics_framework/interface/create_event_handler_udf.py

```diff
@@ -1,10 +1,194 @@
+            event_handler,
+            current_bucketfs_path,
+            latest_bucketfs_path,
+            bucketfs_location)
+
+        # wrap return query if continue else get final_result dictionary
+        return_query_view = None
+        return_query = None
+        final_result = {}
+        query_list = []
+        if isinstance(result, EventHandlerResultFinished):  # result.is_finished
+            final_result = result.final_result
+        else:  # not result.is_finished
+            query_list = result.query_list
+            return_query_view, return_query = self._wrap_return_query(
+                iter_num,
+                bucketfs_connection,
+                self.exa.meta.script_schema,
+                result.return_query)
+
+        # emits
+        ctx.emit(return_query_view)
+        ctx.emit(return_query)
+        ctx.emit(str(result.is_finished))
+        ctx.emit(final_result)
+        for query in query_list:
+            ctx.emit(query)
+
+    def _load_latest_state(
+            self,
+            ctx,
+            iter_num: int,
+            event_handler_class: str,
+            bucketfs_path: PurePosixPath,
+            bucketfs_location: BucketFSLocation) -> EventHandlerState:
+
+        if iter_num > 0:
+            # load the latest state
+            event_handler_state = bucketfs_location.\
+                read_file_from_bucketfs_via_joblib(str(bucketfs_path))
+        else:
+            # create the new state
+            event_handler_module = ctx[3]
+            parameters = ctx[4]
+
+            context = EventHandlerContext(
+                bucketfs_location, bucketfs_path)
+            event_handler_class = getattr(importlib.import_module(
+                event_handler_module), event_handler_class)
+            event_handler_obj = event_handler_class(parameters)
+            event_handler_state = EventHandlerState(
+                context, event_handler_obj, self.get_query_columns())
+
+        return event_handler_state
+
+    @staticmethod
+    def _remove_previous_state(
+            bucketfs_location: BucketFSLocation,
+            bucketfs_path: PurePosixPath) -> None:
+        bucketfs_location.delete_file_in_bucketfs(
+            str(bucketfs_path))
+
+    @staticmethod
+    def _save_current_state(
+            bucketfs_location: BucketFSLocation,
+            bucketfs_path: PurePosixPath,
+            current_state: EventHandlerState) -> None:
+        bucketfs_location.upload_object_to_bucketfs_via_joblib(
+            current_state, str(bucketfs_path))
+
+    def _handle_states(
+            self,
+            event_handler_context: EventHandlerContext,
+            event_handler: EventHandlerBase,
+            current_bucketfs_path: PurePosixPath,
+            latest_bucketfs_path: PurePosixPath,
+            bucketfs_location: BucketFSLocation) -> None:
+
+        current_state = EventHandlerState(
+            event_handler_context, event_handler, self.get_query_columns())
```

Comment:
I think, using self.get_query_columns() doesn't make sense, because self.get_query_columns() create the column from the exa.meta object. The names their are simply the numbers of the columns, because we use a Variadic UDF. I think, here you want to use the columns of the return column and function get_query_columns together with columns of the return query of the previous iteration when creating the EventContext.